### ISSUE TITLE
Build fails

### ISSUE THREAD

**Issue by pol-epsevg** [jturcotte/chiptrack#2]
I was trying to build the desktop application as described with cargo and compilation failed. Here's the relevant part of the error:

```
--- stderr
  error: failed to run custom build command for `wamr-sys v1.2.0 (https://github.com/jturcotte/wamr-sys.git?rev=87680578832f3742d1f4c4f1228ade548bc8a0f0#87680578)`

  CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required):
    Compatibility with CMake < 3.5 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.


  -- Build Configurations:
       Build as target X86_64
       CMAKE_BUILD_TYPE Release
       WAMR Interpreter enabled
       WAMR AOT disabled
       WAMR Fast JIT disabled
       WAMR LLVM ORC JIT disabled
       Libc builtin disabled
       Libc WASI disabled
       Fast interpreter enabled
       Multiple modules disabled
       Bulk memory feature enabled
       Reference types disabled
  wasm-micro-runtime/core/iwasm/include/wasm_export.h:10:10: fatal error: 'stdbool.h' file not found
  thread 'main' panicked at /home/slushee/.cargo/git/checkouts/wamr-sys-6676701d4e58e26a/8768057/build.rs:44:10:
  Unable to generate bindings: ClangDiagnostic("wasm-micro-runtime/core/iwasm/include/wasm_export.h:10:10: fatal error: 'stdbool.h' file not found\n")
```

# Current behavior
Cargo build fails with `'stdbool.h' file not found` on dependency `wamr-sys`

# Expected behavior
Build succeeds and application runs normally

# Enviroment
 - **chiptrack version:** git rev `c15d15e475d3565fffed5ba020a7899e9d5d153b`
 - **rustc version:** `1.75.0`
 - **cc version:** `gcc 12.2.0`
 - **Envoriment:**
   - **OS**: `Void Linux`
   - **Kernel version:** `6.6.11_1` 
   - **Arch:** `x86_64`

**Comment by jturcotte** [comment]
I believe that this file should come with gcc, but it's odd that you'd have the compiler installed without it.

On my Ubuntu 23.10 the file is at `/usr/lib/gcc/x86_64-linux-gnu/12/include/stdbool.h` and is provided by the `libgcc-12-dev` package.

**Comment by pol-epsevg** [comment]
That's weird, because both
 - `/usr/lib/gcc/x86_64-unknown-linux-gnu/12.2/include/stdbool.h` 
 and 
 - `/usr/lib/gcc/x86_64-unknown-linux-gnu/12.2.0/include/stdbool.h` 

definitely exist, have read permissions and contain what you'd expect.

Building on `Fedora 38` with the same kernel, `gcc Red Hat 13.2.1-4` and existing `stdbool.c` files also returns the same error.

I can spin up a few different containers on the cloud to try to build this if you think that would be helpful.

**Comment by jturcotte** [comment]
I verified now by running the `cc` command that is used with gcc's `-H` and it does fetch stdbool.h from that location for me.

I'm wondering if it's trying to use clang for some reason to build on your system, is it possible that clang is installed but not libclang-dev? That would also be weird but there must be some reason...

One thing that could help investigate would be to clone [the wamr-sys crate](https://github.com/jturcotte/wamr-sys) and build it manually to see the output that cargo hides and without all the noise around it:

```sh
git clone git@github.com:jturcotte/wamr-sys.git
cd wamr-sys
git submodule update --init --recursive
mkdir build
cd build
# This step should tell you which compiler is used:
cmake ../libiwasm/CMakeLists.txt
make VERBOSE=1
```

**Comment by pol-epsevg** [comment]
I tried running the build on a VPS to make sure it wasn't because of my machines or something:
 - **OS:** `Ubuntu 23.10`
 - **Arch:** `x86_64 (qemu)`
 - **Kernel:** `6.5.0-9-generic`
 - **cc:** `gcc 13.2.0-4ubuntu3`
 
I needed to install these packages for the compilation to succeed on the VPS: `gcc g++ clang pkg-config cmake alsa libasound2-dev libxft-dev` (maybe you could add these to the README) 

Installing clang on the VPS made me think that maybe I needed it on `Void` too. And that was exactly it. After installing `clang`, the build succeeded on my machine. Thanks for the patience.

## Stuff that's now irrelevant
I don't have `clang` installed, and it would be really weird for cargo to not use `cc`. Cargo also succeeds on building `cc` and `libc` when trying to compile this repo.

`jturcotte/wamr-sys` seemed to succeed compilation with `[100%] Built target iwasm`, telling me it's using `/usr/bin/cc`

`cc -print-libgcc-file-name` tells me it's using `/usr/lib64/gcc/x86_64-unknown-linux-gnu/12.2.0/`, where `include/stdbool.h` exists

By the way, cargo says you've got an unused import:
```rs
warning: unused import: `SoundRenderer`
 --> src/sound_renderer.rs:7:40
  |
7 | pub use emulated::{new_sound_renderer, SoundRenderer, Synth};
  |                                        ^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
```

Feel free to close the issue. If you want me to make a quick PR adding the dependencies to the README and removing the unused import let me know

**Comment by jturcotte** [comment]
Thanks for the update. I tried installing clang locally and then ran `cargo build -vv` to check the output and cmake is still picking up gcc for me, uninstalling it also works fine. So the only explanation I have is that maybe some environment variable on your system could instruct it otherwise, and a missing libclang could lead to this situation.

I've added some missing dependencies to the README. I've added a github action to check the Ubuntu build and only `libasound2-dev` oddly still needed to be added there. [The image already seems to include build-essential, pkg-config and cmake](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md), but I'm not sure if libxft-dev is there. I left out clang either way as I think it should normally not be necessary.

For the next release I'm planing to at least add macOS and Game Boy Advance binaries, plus maybe Windows as well. So that should at least prevent non-developer users on those operating systems from having to deal with this.
